### PR TITLE
#1279: run modelling validations on node <assetname> only

### PR DESCRIPTION
**Pull request by aparna-d2**
Also, retain only node <assetname> in published file (delete rest), but keep entire contents in work file while versioning up.

### CODE REVIEW

File: hooks/tk-multi-publish2/maya/publish_session_model.py [hunk 1]
```diff
@@ -59,16 +61,45 @@ def validate(self, task_settings, item):
         """
         # create list of elements from maya nodes to collect results about
         workflow_data = {"elements": []}
+        # TODO: replace with find_model_root_nodes from modelpublish-8.7.0+
         toplevel_objects = findModelRootNodes()
+        valid_node_name = item.context.entity['name'].lower()
```
Comment (aparna-d2) [inline hooks/tk-multi-publish2/maya/publish_session_model.py]: Waiting to determine whether we should use `entity['name']` or `task['name']` depending on the department's workflow (creating subassets called x or creating task x within the asset itself)

File: hooks/tk-multi-publish2/maya/publish_session_model.py [hunk 2]
```diff
@@ -99,3 +131,46 @@ def validate(self, task_settings, item):
...
+        :param publish_path: The output path to publish files to
+        """
+
+        path = copy.deepcopy(item.properties.get("path"))
+        if not path:
+            raise KeyError("Base class implementation of publish_files() method requires a 'path' property.")
+
+        # Save to publish path
+        self.cleanup_file(item)
+        self._save_session(publish_path, item)
```
Comment (mbrainerd) [inline hooks/tk-multi-publish2/maya/publish_session_model.py]: FYI, the new version of `self._save_session()` also requires a `version` argument.

Comment (mbrainerd) [reply hooks/tk-multi-publish2/maya/publish_session_model.py]: sgtk_apps-3.18.0_bug1258

**Comment by garvitverma** [comment]
Merging to master now, since we have bug_1258 branch of tk-multi-publish2 released.

**Pull request merged**